tensorflow/core/kernels/cudnn_rnn_ops.cc

```diff
@@ -635,6 +655,46 @@ Status CreateForwardAndBackwardIODescriptors(
       output_shape.dim_size(2), data_type);
   TF_RETURN_IF_ERROR(output_desc_s.status());
   *output_desc = output_desc_s.ConsumeValueOrDie();
+
+  return Status::OK();
+}
+
+template <typename T>
+Status CreateForwardAndBackwardIODescriptors(
```

Comment:
Alright I pushed this too far.

I'm ok to have one unified `CreateForwardAndBackwardIODescriptors` and `DoForward` and `DoBackward` which accept `absl::Span<int> seq_lengths`, and in their bodies they'll act appropriately based on if the `span` is nullptr or not.